pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/json/BaseJsonIndexCreator.java

```diff
@@ -91,7 +98,36 @@ public abstract class BaseJsonIndexCreator implements JsonIndexCreator {
   @Override
   public void add(String jsonString)
       throws IOException {
-    addFlattenedRecords(JsonUtils.flatten(jsonString, _jsonIndexConfig));
+    List<Map<String, String>> flattenedRecord = JsonUtils.flatten(jsonString, _jsonIndexConfig);
+    if (flattenedRecord.equals(JsonUtils.SKIPPED_FLATTENED_RECORD)) {
```

Comment:
@gortiz @Jackie-Jiang just FYI that I've added additional code to check if `JsonUtils.flatten` returned the SKIPPED record and I update the metric in that case.